Source/Core/Runtime/StateMachineRuntime.cs

```diff
@@ -207,6 +237,17 @@ public override void RemoteSendEvent(MachineId target, Event e, Guid? operationG
         }
 
         /// <summary>
+        /// Create a new machine Id with incremented generation. This
+        /// method should be used only for testing with PSharpTester.
+        /// </summary>
+        /// <param name="mid">Machine Id</param>
+        /// <returns>Machine Id with incremented generation count</returns>
+        public override MachineId IncGenerationForTesting(MachineId mid)
+        {
+            throw new NotImplementedException("The method IncGenerationForTesting should not be used for production runs.");
```

Comment:
I think this should change to `NotSupportedException`, to make it 100% clear that we do not support this method in the production runtime (versus that it is not implemented now, but might be in the future).